File: play-walletless-games-front-end/src/components/Game.js
import {useEffect, useState} from "react";
import {useParams} from "react-router-dom";

const Game = () => {
    const [game, setGame] = useState({});
    let { id } = useParams();

    useEffect(() => {
         let game = {
             id: 1,
             title: "The Legend of Zelda: Breath of the Wild",
             release_date: "2017",
             esrb_rating: "E10+",
             genre: "Action-Adventure",
             description: "Sample description"
          };
          setGame(game)
    }, [id])

    return(
        <div>
            <h2>Game: {game.title}</h2>
            <small><em>{game.release_date}, Genre: {game.genre}, Rating: {game.esrb_rating}</em></small>
            <hr/>
            <p>{game.description}</p>
        </div>
    )
}

export default Game;

File: play-walletless-games-back-end/internal/models/Game.go
package models

import "time"

type Game struct {
	ID          int       `json:"id"`
	Title       string    `json:"title"`
	ReleaseDate time.Time `json:"release_date"`
	ESRBRating  string    `json:"esrb_rating"`
	Genre       string    `json:"genre"`
	Description string    `json:"description"`
	Image       string    `json:"image"`
	CreatedAt   time.Time `json:"-"`
	UpdatedAt   time.Time `json:"-"`
}
